### PR TITLE
Install gh package from git-cli

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description

gh package from git-cli repo is added to the dependencies list

### Issues Resolved
This PR enables the use of gh package which is used to create Github issue when a component fails while running integtests in https://github.com/opensearch-project/opensearch-build/pull/3465

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker/ci/dockerfiles/current/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile [hunk 1]
```diff
@@ -18,6 +18,11 @@ RUN yum clean all && \
     yum update -y && \
     yum install -y which curl git gnupg2 tar net-tools procps-ng python3 python3-devel python3-pip zip unzip
 
+# Install gh cli
+RUN dnf clean all && dnf install -y 'dnf-command(config-manager)' && dnf config-manager --add-repo https://cli.github.com/packages/rpm/gh-cli.repo && \
+    dnf update -y && \
+    dnf install -y which curl git gnupg2 tar net-tools procps-ng python3 python3-devel python3-pip zip unzip jq gh
```
Comment (gaiksaya) [inline docker/ci/dockerfiles/current/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile:24]: We don't need all those tools just `gh`
```suggestion
    dnf install -y gh
```

File: docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile [hunk 2]
```diff
@@ -23,6 +23,11 @@ RUN dnf clean all && \
     dnf update -y && \
     dnf install -y which curl git gnupg2 tar net-tools procps-ng python3 python3-devel python3-pip zip unzip jq
 
+# Install gh cli
+RUN dnf clean all && dnf install -y 'dnf-command(config-manager)' && dnf config-manager --add-repo https://cli.github.com/packages/rpm/gh-cli.repo && \
+    dnf update -y && \
+    dnf install -y which curl git gnupg2 tar net-tools procps-ng python3 python3-devel python3-pip zip unzip jq gh
```
Comment (gaiksaya) [inline docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile:29]:
```suggestion
    dnf install -y gh
```

**Review by gaiksaya — COMMENTED** [review]
Just install `gh` nothing else. Also can you add more details in description such as installing gh on docker images required for testing dashboards components? Fix the PR title too

File: docker/ci/dockerfiles/current/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile (hunk 1 above)
Comment (Divyaasm) [reply docker/ci/dockerfiles/current/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile:24]: Sure Sayali

File: docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile (hunk 2 above)
Comment (Divyaasm) [reply docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile:29]: And one more question @gaiksaya , I see on of the images uses `yum` and the other uses `dnf` can I use the similar command to install gh. But both the installations are working

**Comment by peterzhuamazon** [comment]
> Just install `gh` nothing else. Also can you add more details in description such as installing gh on docker images required for testing dashboards components? Fix the PR title too

The ubuntu image already have gh.
We can actually add gh for all docker images if needed.

Thanks.

File: docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile (hunk 2 above)
Comment (gaiksaya) [reply docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile:29]: Whatever is easier. @peterzhuamazon Any preference?

Comment (Divyaasm) [reply docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile:29]: > > Just install `gh` nothing else. Also can you add more details in description such as installing gh on docker images required for testing dashboards components? Fix the PR title too
> 
> The ubuntu image already have gh. We can actually add gh for all docker images if needed.
> 
> Thanks.

Added changes only to two images(rockylinux8) lacking gh used by OSD , I see there are two more docker files that do not have gh package, should I update them too?

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile:24]: > 

yum is just a wrapper/symlink of dnf on RHEL8+, either is ok but certain commands relied on dnf, use dnf here is ok imo.

Comment (peterzhuamazon) [reply docker/ci/dockerfiles/current/test.rockylinux8.systemd-base.x64.arm64.dockerfile:24]: > > > Just install `gh` nothing else. Also can you add more details in description such as installing gh on docker images required for testing dashboards components? Fix the PR title too
> > 
> > 
> > The ubuntu image already have gh. We can actually add gh for all docker images if needed.
> > Thanks.
> 
> Added changes only to two images(rockylinux8) lacking gh used by OSD , I see there are two more docker files that do not have gh package, should I update them too?

Please.

**Review by peterzhuamazon — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
Workflows:
https://build.ci.opensearch.org/job/docker-build/3335/console
https://build.ci.opensearch.org/job/docker-build/3336/console
https://build.ci.opensearch.org/job/docker-build/3337/console
https://build.ci.opensearch.org/job/docker-build/3338/console

https://build.ci.opensearch.org/job/docker-copy/452/console

**Pull request merged**